File: a2q1b.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    float m,n,o,p, i=0;
    int N = 6;
    float d =0;
    for (m=0; m<N; m++)
        for(o=0; o<N;o++)
    {
        for (n=0; n<N; n++)
            for(p=0; p<N;p++)
        {
            d+=abs(m-n)+abs(o-p);
            i++;
        
        }
    };
    float k=d/i;
    
    printf("The average distance  %f",k);
    return 0;
}

//OUTPUT
//The average distance : 3.888889

File: a2q1b.py
#define each 
prompt1 = "number of equidistant points in line:"
N=6
#N = int(input(prompt1))

arr=[]
#(x1,y1) ~ (m,o) & (x2,y2) ~ (n,p)
#looping 
for m in range(0,N):
	for o in range(0,N):
		for n in range (0,N):
			for p in range(0,N):
				#calculating distance 
				d = abs(m-n)+abs(o-p)
				#store it in array 
				arr.append(d)
	

#to check if all datas are intact we can print the array
#print(arr)
#print(len(arr))
#print(sum(arr))
print(f'Average distance : {sum(arr)/len(arr)}')


#Average distance : 3.888888888888889